lyb: warnings when building for ARM 32bit

```
/home/jkt/work/prog/_build/br-cfb/build/libyang-custom/src/parser_lyb.c: In function ‘lyb_parse_val_2’:
/home/jkt/work/prog/_build/br-cfb/build/libyang-custom/src/parser_lyb.c:569:29: warning: format ‘%ld’ expects argument of type ‘long int’, but argument 3 has type ‘int64_t {aka long long int}’ [-Wformat=]
         sprintf(num_str, "%ld", value->int64);
                           ~~^   ~~~~~~~~~~~~
                           %lld
/home/jkt/work/prog/_build/br-cfb/build/libyang-custom/src/parser_lyb.c:573:29: warning: format ‘%lu’ expects argument of type ‘long unsigned int’, but argument 3 has type ‘uint64_t {aka long long unsigned int}’ [-Wformat=]
         sprintf(num_str, "%lu", value->uint64);
                           ~~^   ~~~~~~~~~~~~~
                           %llu
/home/jkt/work/prog/_build/br-cfb/build/libyang-custom/src/parser_lyb.c:585:29: warning: format ‘%ld’ expects argument of type ‘long int’, but argument 3 has type ‘long long int’ [-Wformat=]
         sprintf(num_str, "%ld.%.*ld", value->dec64 / (int64_t)type->info.dec64.div, dig, frac);
                           ~~^         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                           %lld
/home/jkt/work/prog/_build/br-cfb/build/libyang-custom/src/parser_lyb.c:585:35: warning: format ‘%ld’ expects argument of type ‘long int’, but argument 5 has type ‘int64_t {aka long long int}’ [-Wformat=]
         sprintf(num_str, "%ld.%.*ld", value->dec64 / (int64_t)type->info.dec64.div, dig, frac);
                               ~~~~^
                               %.*lld
```

Thanks, these are OK now. What remains are some smaller warnings which (I suppose) are tricky to fix in a portable manner):
```
src/parser_json.c: In function ‘lyjson_parse_text’:
src/parser_json.c:153:32: warning: comparison is always true due to limited range of data type [-Wtype-limits]
         } else if ((data[*len] >= 0 && data[*len] < 0x20) || data[*len] == 0x5c) {
                                ^~
src/printer_json.c: In function ‘json_print_string’:
src/printer_json.c:44:21: warning: comparison is always true due to limited range of data type [-Wtype-limits]
         if (text[i] >= 0 && text[i] < 0x20) {
                     ^~
linenoise/linenoise.c: In function ‘linenoiseEdit’:
linenoise/linenoise.c:885:19: warning: comparison is always false due to limited range of data type [-Wtype-limits]
             if (c < 0) return ls.len;
                   ^
```
Feel free to close this bug, these warnings look harmless, unlike the already fixed ones.

--- a/src/parser_json.c
+++ b/src/parser_json.c
@@ -150,7 +150,7 @@ lyjson_parse_text(struct ly_ctx *ctx, const char *data, unsigned int *len)
             }
             o += r - 1; /* o is ++ in for loop */
             (*len) += i; /* number of read characters */
-        } else if ((data[*len] >= 0 && data[*len] < 0x20) || data[*len] == 0x5c) {
+        } else if ((((signed char)data[*len]) >= 0 && data[*len] < 0x20) || data[*len] == 0x5c) {
             /* control characters must be escaped */
             LOGVAL(ctx, LYE_XML_INVAL, LY_VLOG_NONE, NULL, "control character (unescaped)");
             goto error;

--- a/src/printer_json.c
+++ b/src/printer_json.c
@@ -41,7 +41,7 @@ json_print_string(struct lyout *out, const char *text)
 
     ly_write(out, "\"", 1);
     for (i = n = 0; text[i]; i++) {
-        if (text[i] >= 0 && text[i] < 0x20) {
+        if (((signed char)text[i]) >= 0 && text[i] < 0x20) {
             /* control character */
             n += ly_print(out, "\\u%.4X", text[i]);
         } else {

